src/app/components/image-viewer/ImageViewer.tsx

```diff
@@ -1,27 +1,56 @@
 /* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import FileSaver from 'file-saver';
 import classNames from 'classnames';
 import { Box, Chip, Header, Icon, IconButton, Icons, Text, as } from 'folds';
 import * as css from './ImageViewer.css';
 import { useZoom } from '../../hooks/useZoom';
-import { usePan } from '../../hooks/usePan';
+import { ImageController } from './ImageController';
+import { Pan } from '../../hooks/usePan';
 
 export type ImageViewerProps = {
   alt: string;
   src: string;
   requestClose: () => void;
 };
 
+const INITIAL_PAN: Pan = {
+  translateX: 0,
+  translateY: 0,
+};
+
 export const ImageViewer = as<'div', ImageViewerProps>(
   ({ className, alt, src, requestClose, ...props }, ref) => {
-    const { zoom, zoomIn, zoomOut, setZoom } = useZoom(0.2);
-    const { pan, cursor, onMouseDown } = usePan(zoom !== 1);
+    const [pan, setPan] = useState(INITIAL_PAN);
+    const { zoom, zoomIn, zoomOut, setZoom } = useZoom(0.2, 0.2, 6);
+
+    const zoomOutOnClick = useCallback(() => {
+      const newZoom = zoomOut();
+
+      setPan({
+        translateX: (pan.translateX * newZoom) / zoom,
+        translateY: (pan.translateY * newZoom) / zoom,
```

Comment:
I am not sure what kind of transformation is applied on data before assigning to `translateX` & `Y`. You can move this to a named utility function.